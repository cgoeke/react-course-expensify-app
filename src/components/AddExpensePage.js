import React, { Component } from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';

export const AddExpensePage = (props) => (
    <div>
        <h1>Add Expense Component</h1>
        <ExpenseForm 
            onSubmit={(expense) => {
                props.startAddExpense(expense);
                props.history.push('/');
            }}
        />
    </div>
);

const mapDispatchToProps = (dispatch) => {
    return {
        startAddExpense: (expense) => dispatch(startAddExpense(expense))
    };
};

export default connect(undefined, mapDispatchToProps)(AddExpensePage);