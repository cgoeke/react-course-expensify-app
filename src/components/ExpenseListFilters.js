import React, { Component } from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';
import 'react-dates/initialize';

export class ExpenseListFilters extends Component {
    constructor(props) {
        super(props);

        this.state = {
            focused: null
        }

        this.onDatesChange = this.onDatesChange.bind(this);
        this.onFocusChange = this.onFocusChange.bind(this);
    };

    onDatesChange({ startDate, endDate }) {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };

    onFocusChange(focused) {
        this.setState(() => ({ focused }));
    };

    render() {
      return (
        <div className="content-container">
          <div className="input-group">
            <div className="input-group__item">
              <input 
                type="text" 
                className="text-input"
                placeholder="Search Expenses"
                value={this.props.filters.text}
                onChange={(e) => {
                  this.props.setTextFilter(e.target.value);
                }} 
              />
            </div>
            <div className="input-group__item">
              <select
                className="select"
                value={this.props.filters.sortBy}
                onChange={(e) => {
                  e.target.value === 'date' ?
                    this.props.sortByDate(e.target.value) :
                    this.props.sortByAmount(e.target.value)
                }}
              >
                <option value="date">Date</option>
                <option value="amount">Amount</option>
              </select>
            </div>
            <div className="input-group__item">
              <DateRangePicker 
                startDateId='start'
                startDate={this.props.filters.startDate}
                endDateId='end'
                endDate={this.props.filters.endDate} 
                onDatesChange={this.onDatesChange}                
                focusedInput={this.state.focused} 
                onFocusChange={this.onFocusChange}
                showClearDates={true}
                numberOfMonths={1}
                isOutsideRange={() => false}
              />
            </div>
          </div>
        </div>
        );
    };
};

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        setTextFilter: (text) => dispatch(setTextFilter(text)),
        setStartDate: (startDate) => dispatch(setStartDate(startDate)),
        setEndDate: (endDate) => dispatch(setEndDate(endDate)),
        sortByDate: () => dispatch(sortByDate()),
        sortByAmount: () => dispatch(sortByAmount())
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);