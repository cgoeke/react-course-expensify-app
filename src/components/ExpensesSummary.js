import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
    return (
        <div>
            {expenseCount > 0 &&
                <p>{`Viewing ${expenseCount} ${expenseCount === 1 ? 'expense' : 'expenses'} totalling ${formattedExpensesTotal}`}</p>
            }
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
  };

export default connect(mapStateToProps)(ExpensesSummary);