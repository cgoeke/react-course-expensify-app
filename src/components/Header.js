import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="">Create Expense</NavLink>
        <NavLink to="/help" activeClassName="">Help</NavLink>
    </header>
); 

export default Header;