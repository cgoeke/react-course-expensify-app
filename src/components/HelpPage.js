import React, { Component } from 'react';

const HelpPage = () => (
    <h1>Help Component</h1>
);

export default HelpPage;