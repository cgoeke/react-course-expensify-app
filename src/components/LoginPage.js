import React, { Component } from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
      <div className="login">
        <div className="login__box">
          <h1 className="login__title">Expensify</h1>
          <p>Keep track of your expenses</p>
          <button className="button" 
            onClick={startLogin} 
            type="submit">
              Login with Google
          </button>
        </div>
      </div>
    );

const mapDispatchToProps = (dispatch) => {
  return {
    startLogin: () => dispatch(startLogin())
  };
};

export default connect(undefined, mapDispatchToProps)(LoginPage);