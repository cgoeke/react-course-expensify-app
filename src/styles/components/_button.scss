.button {
  background: $blue;
  display: inline-block;
  color: $white;
  font-weight: 300;
  font-size: $font-size-l;
  padding: $s-size $m-size;
  border: none;
  line-height: 1;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888888;
}