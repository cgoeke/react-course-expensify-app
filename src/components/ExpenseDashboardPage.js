import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashboardPage = () => (
    <div>
        <h1>Expense Dashboard Component</h1>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;