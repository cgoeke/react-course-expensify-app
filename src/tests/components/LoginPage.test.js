import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('renders LoginPage component', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test('calls startLogin when button is clicked', () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLogin} />);
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});