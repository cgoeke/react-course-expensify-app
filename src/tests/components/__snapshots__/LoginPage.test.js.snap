// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoginPage component 1`] = `
<div
  className="login"
>
  <div
    className="login__box"
  >
    <h1
      className="login__title"
    >
      Expensify
    </h1>
    <p>
      Keep track of your expenses
    </p>
    <button
      className="button"
      type="submit"
    >
      Login with Google
    </button>
  </div>
</div>
`;
